gnocchi/storage/__init__.py

```diff
@@ -199,35 +200,55 @@ def _version_check(name, v):
         """
         return name.split("_")[-1] == 'v%s' % v
 
+    def get_aggregated_measures(self, metric, aggregations,
+                                from_timestamp=None, to_timestamp=None):
+        """Get aggregated measures from a metric.
+
+        :param metric: The metric measured.
+        :param aggregations: The aggregations to retrieve.
+        :param from timestamp: The timestamp to get the measure from.
+        :param to timestamp: The timestamp to get the measure to.
+        """
+        timeseries = utils.parallel_map(
+            self._get_measures_timeserie,
+            ((metric, agg, from_timestamp, to_timestamp)
+             for agg in aggregations))
+        return {
+            agg: ts.fetch(from_timestamp, to_timestamp)
+            for agg, ts in six.moves.zip(aggregations, timeseries)
+        }
+
     def get_measures(self, metric, aggregations,
                      from_timestamp=None, to_timestamp=None,
                      resample=None):
-        """Get a measure to a metric.
+        """Get aggregated measures from a metric.
+
+        Deprecated. Use `get_aggregated_measures` instead.
 
         :param metric: The metric measured.
         :param aggregations: The aggregations to retrieve.
         :param from timestamp: The timestamp to get the measure from.
         :param to timestamp: The timestamp to get the measure to.
         :param resample: The granularity to resample to.
         """
-        agg_timeseries = utils.parallel_map(
-            self._get_measures_timeserie,
-            ((metric, ag, from_timestamp, to_timestamp)
-             for ag in aggregations))
+        timeseries = self.get_aggregated_measures(
+            metric, aggregations,
+            from_timestamp=None, to_timestamp=None)
 
         if resample:
-            agg_timeseries = list(map(lambda agg: agg.resample(resample),
-                                      agg_timeseries))
+            for agg, ts in six.iteritems(timeseries):
+                timeseries[agg] = ts.resample(resample)
 
         return {
             aggmethod: list(itertools.chain(
-                *[[(timestamp, ts.aggregation.granularity, value)
+                *[[(timestamp, timeseries[agg].aggregation.granularity, value)
                    for timestamp, value
-                   in ts.fetch(from_timestamp, to_timestamp)]
-                  for ts in aggts]))
-            for aggmethod, aggts
-            in itertools.groupby(agg_timeseries,
-                                 lambda v: v.aggregation.method)
+                   in timeseries[agg].fetch(from_timestamp, to_timestamp)]
+                  for agg in sorted(aggs,
+                                    key=ATTRGETTER_GRANULARITY,
+                                    reverse=True)]))
+            for aggmethod, aggs in itertools.groupby(timeseries.keys(),
+                                                     ATTRGETTER_METHOD)
```

Comment:
tbh, this is some crazy looking looping but i'm guessing you're going to remove this later on?

Reply:
It's the same as before and that's what the API returns actually. It should be a bit simpler when this method gets entirely removed and moved to the API>

Reply:
works for me.